make/framework.mk

```diff
@@ -128,6 +129,7 @@ test_warnings: $(FRAMEWORK_HEADER)
 	@rm $(FRAMEWORK_HEADER:%.h=%.o)
 
 $(MODULE_MAP):
+	@mkdir -p $$(dirname $@)
 	@echo "module" $(FRAMEWORK_NAME) "{" > $(MODULE_MAP)
```

Comment:
I was tempted to change all occurrences of `$(MODULE_MAP)` in the `$(MODULE_MAP)` target's commands to `$@` but I decided against it. I'll leave that for a follow-up pull request in order to keep this one focused on the change at hand.